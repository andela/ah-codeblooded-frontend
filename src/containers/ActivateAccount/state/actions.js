import {
  ACTIVATE_ERROR,
  ACTIVATE_SUCCESS,
  ACTIVATE_USER,
} from './types';
import api from '../../../utils/api';

export const activateUser = () => ({
  type: ACTIVATE_USER,
});

export const activationSuccess = () => ({
  type: ACTIVATE_SUCCESS,
  payload: { msg: 'Account activation was successful' },
});

export const activationError = msg => ({
  type: ACTIVATE_ERROR,
  payload: { msg },
});


const activateUserAction = (token, uid) => (dispatch) => {
  dispatch(activateUser());
  return api.get(`account/verify/${token}/${uid}/`).then(() => {
    dispatch(activationSuccess());
  }).catch((error) => {
    const msg = error.response && "Something went wrong. Try again";
    dispatch(activationError(msg));
  });
};

export default activateUserAction;
