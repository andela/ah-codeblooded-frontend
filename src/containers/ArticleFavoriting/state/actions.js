import api from '../../../utils/api';

import { FAVORITE_ARTICLE_SUCCESS, FAVORITE_UNFAVORITE_ERROR } from './types';

export const favoriteArticleSuccess = payload => ({
  type: FAVORITE_ARTICLE_SUCCESS,
  payload,
});

export const favoriteUnfavoriteFailure = errors => ({
  type: FAVORITE_UNFAVORITE_ERROR,
  errors,
});

export const favoriteArticleAction = (slug, favourited) => (dispatch) => {
  const url = `articles/${slug}/favourite/`;
  const axios = favourited ? api.delete(url) : api.post(url);
  return axios
    .then(res => dispatch(favoriteArticleSuccess(res.data.favourited)))
    .catch(errors => dispatch(favoriteUnfavoriteFailure(errors)));
};
