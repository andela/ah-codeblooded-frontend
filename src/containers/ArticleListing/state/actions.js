import api from '../../../utils/api';
import {
  FETCH_ARTICLES,
  FETCH_ARTICLES_FAILURE,
  FETCH_ARTICLES_SUCCESS,
  FETCH_MORE_ARTICLES_SUCCESS,
} from './types';

export const fetchingArticles = listName => (
  { type: FETCH_ARTICLES, payload: { listName } }
);

export const articlesFetchingSuccessful = (articles, listName, fetchMore) => (
  {
    type: fetchMore ? FETCH_MORE_ARTICLES_SUCCESS : FETCH_ARTICLES_SUCCESS,
    payload: { data: articles, listName },
  }
);

export const articlesFetchingFailed = (errors, listName) => (
  { type: FETCH_ARTICLES_FAILURE, payload: { errors, listName } }
);

export const articlesFetchAction = (url, params, listName, fetchMore = false) => (dispatch) => {
  dispatch(fetchingArticles(listName));
  return api.get(url, { params })
    .then((response) => {
      dispatch(articlesFetchingSuccessful(response.data.data.article, listName, fetchMore));
    }).catch((error) => {
      dispatch(articlesFetchingFailed(error.response, listName));
    });
};
