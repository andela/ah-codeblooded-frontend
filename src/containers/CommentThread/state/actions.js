import {
  DELETING_COMMENT,
  DELETING_COMMENT_FAILED,
  DELETING_COMMENT_SUCCESSFUL,
  FETCHING_COMMENTS,
  FETCHING_COMMENTS_FAILED,
  FETCHING_COMMENTS_SUCCESS,
  FETCHING_MORE_COMMENTS,
  FETCHING_MORE_COMMENTS_FAILED,
  FETCHING_MORE_COMMENTS_SUCCESS,
} from './types';
import api from '../../../utils/api';

export const deletingComment = () => ({
  type: DELETING_COMMENT,
});

export const deletingCommentSuccessful = () => ({
  type: DELETING_COMMENT_SUCCESSFUL,
});

export const deletingCommentFailed = () => ({
  type: DELETING_COMMENT_FAILED,
});

export const fetchingComments = (parentId, page = 1) => ({
  type: page === 1 ? FETCHING_COMMENTS : FETCHING_MORE_COMMENTS,
  payload: { parent: parentId },
});

export const fetchingCommentsSuccess = (data, parentId, page = 1) => ({
  type: page === 1 ? FETCHING_COMMENTS_SUCCESS : FETCHING_MORE_COMMENTS_SUCCESS,
  payload: { data, parent: parentId },
});

export const fetchingCommentsFailed = (errors, parentId, page = 1) => ({
  type: page === 1 ? FETCHING_COMMENTS_FAILED : FETCHING_MORE_COMMENTS_FAILED,
  payload: { data: errors, parent: parentId },
});

export const fetchCommentsAction = (slug, parentId = null, page = 1) => (dispatch) => {
  dispatch(fetchingComments(parentId, page));
  return api.get(`articles/${slug}/comments${parentId ? `/${parentId}` : ''}`, { params: { page } })
    .then((response) => {
      dispatch(fetchingCommentsSuccess(response.data.data.comment, parentId, page));
    }).catch((error) => {
      dispatch(fetchingCommentsFailed(error.response, parentId, page));
    });
};

export const deleteCommentAction = (slug, parentId, commentId, successCallback) => (dispatch) => {
  dispatch(deletingComment());
  return api.delete(`articles/${slug}/comments/${commentId}`).then(() => {
    dispatch(deletingCommentSuccessful());
    dispatch(fetchCommentsAction(slug, parentId));
    successCallback();
  }).catch(() => {
    dispatch(deletingCommentFailed());
  });
};
