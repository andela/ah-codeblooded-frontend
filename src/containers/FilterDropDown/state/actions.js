import api from '../../../utils/api';
import { FETCH_AUTHORS, FETCH_TAGS } from './types';

export const fetchAuthorsAction = () => dispatch => api.get('users/search/')
  .then((response) => {
    dispatch({ type: FETCH_AUTHORS, payload: response.data });
  });

export const fetchTagsAction = () => dispatch => api.get('tags/')
  .then((response) => {
    dispatch({ type: FETCH_TAGS, payload: response.data });
  });
