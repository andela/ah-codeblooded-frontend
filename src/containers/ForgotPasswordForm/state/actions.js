import {
  FORGOT_PASSWORD_REQUEST,
  FORGOT_PASSWORD_SUCCESS,
  FORGOT_PASSWORD_FAIL,
  RESEND_RESET_LINK,
} from './types';
import api from '../../../utils/api';

export const forgotPasswordAction = email => (dispatch) => {
  dispatch({
    type: FORGOT_PASSWORD_REQUEST,
  });
  return api.post('/account/forgot_password/', { email })
    .then((res) => {
      dispatch({
        type: FORGOT_PASSWORD_SUCCESS,
        payload: { message: res.data.message },
      });
    }).catch((err) => {
      const message = err.response ? err.response.data.message : 'Something went wrong. Try again.';
      dispatch({
        type: FORGOT_PASSWORD_FAIL,
        payload: { error: message },
      });
    });
};

export const resendLinkAction = () => dispatch => (
  dispatch({
    type: RESEND_RESET_LINK,
  })
);
