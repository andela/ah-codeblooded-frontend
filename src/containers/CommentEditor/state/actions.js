import {
  COMMENTING,
  COMMENTING_FAILED,
  COMMENTING_SUCCESS, FETCHING_COMMENT_AUTHORS_SUCCESS,
  FETCHING_COMMENT_AUTHORS, UPDATING_COMMENT, UPDATING_COMMENT_SUCCESS, UPDATING_COMMENT_FAILED,
} from './types';
import api from '../../../utils/api';
import { fetchCommentsAction } from '../../CommentThread/state/actions';

export const updatingComment = () => ({
  type: UPDATING_COMMENT,
});

export const updatingCommentSuccess = () => ({
  type: UPDATING_COMMENT_SUCCESS,
});

export const updatingCommentFailed = () => ({
  type: UPDATING_COMMENT_FAILED,
});

export const fetchingCommentAuthors = () => ({
  type: FETCHING_COMMENT_AUTHORS,
});

export const fetchingCommentAuthorsSuccess = authors => ({
  type: FETCHING_COMMENT_AUTHORS_SUCCESS,
  payload: authors,
});

export const commentingAction = () => ({
  type: COMMENTING,
});

export const commentingSuccess = comment => ({
  type: COMMENTING_SUCCESS,
  payload: comment,
});

export const commentingFailed = errors => ({
  type: COMMENTING_FAILED,
  payload: errors,
});

export const fetchCommentAuthorsAction = slug => (dispatch) => {
  dispatch(fetchingCommentAuthors());
  return api.get(`articles/${slug}/comments/authors`)
    .then((response) => {
      dispatch(fetchingCommentAuthorsSuccess(response.data.data.users));
    });
};

export const updateCommentAction = (slug, comment,
  commentId, mentions, successCallback) => (dispatch) => {
  dispatch(updatingComment());
  return api.put(`articles/${slug}/comments/${commentId}`, {
    comment:
      { body: comment },
    mentions,
  })
    .then((response) => {
      dispatch(updatingCommentSuccess());
      successCallback(response.data.data.comment);
      dispatch(fetchCommentsAction(slug, response.data.data.comment.parent));
    }).catch(() => {
      dispatch(updatingCommentFailed());
    });
};

export const commentAction = (
  slug,
  comment,
  commentId = null,
  mentions,
  successCallback,
) => (dispatch) => {
  dispatch(commentingAction());
  return api.post(`articles/${slug}/comments${commentId ? `/${commentId}` : ''}`,
    { comment: { body: comment }, mentions })
    .then((response) => {
      dispatch(commentingSuccess(response.data.data.comment));
      successCallback();
      dispatch(fetchCommentsAction(slug, commentId));
    }).catch((response) => {
      dispatch(commentingFailed(response.response));
    });
};
