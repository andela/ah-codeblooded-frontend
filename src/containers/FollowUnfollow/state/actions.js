import api from "../../../utils/api";
import {
  FOLLOW_USER_FAILURE, FOLLOW_USER_SUCCESS, UNFOLLOW_USER_FAILURE, UNFOLLOW_USER_SUCCESS,
  FETCH_FOLLOWERS_SUCCESS, FETCH_FOLLOWERS_FAILURE,
} from "./type";
import listTypes from '../../UsersListing/state/listTypes';
import {
  pageLoadingAction,
  pageLoadedAction,
} from "../../NavBar/state/actions";
import { fetchUsersAction } from "../../UsersListing/state/actions";
import { getCurrentUser } from "../../../utils/auth";

export const checkFollowers = payload => ({
  type: FETCH_FOLLOWERS_SUCCESS,
  payload,
});

export const checkFollowingFailure = error => ({
  type: FETCH_FOLLOWERS_FAILURE,
  error,
});

export const followSuccess = payload => ({
  type: FOLLOW_USER_SUCCESS,
  payload,
});

export const followFailure = error => ({
  type: FOLLOW_USER_FAILURE,
  error,
});

export const UnfollowSuccess = payload => ({
  type: UNFOLLOW_USER_SUCCESS,
  payload,
});

export const UnfollowFailure = error => ({
  type: UNFOLLOW_USER_FAILURE,
  error,
});

export const fetchAction = username => (dispatch) => {
  dispatch(pageLoadingAction());
  return api.get(`profiles/following/${username}`)
    .then((res) => {
      const following = res.data.following_status;
      if (following) {
        dispatch(checkFollowers(res.data));
        dispatch(pageLoadedAction());
      }
    }).catch((error) => {
      dispatch(checkFollowingFailure(error));
      dispatch(pageLoadedAction());
    });
};

export const followAction = username => (dispatch) => {
  dispatch(pageLoadingAction());
  const user = getCurrentUser();
  return api
    .post(`profiles/${username}/follow/`)
    .then((res) => {
      dispatch(followSuccess(res.data));
      dispatch(pageLoadedAction());
      dispatch(fetchUsersAction(user.username, listTypes.FOLLOWING));
      dispatch(fetchUsersAction(user.username, listTypes.FOLLOWERS));
    })
    .catch((errors) => {
      dispatch(followFailure(errors.response));
      dispatch(pageLoadedAction());
    });
};

export const unfollowAction = username => (dispatch) => {
  const user = getCurrentUser();
  return api.delete(`profiles/${username}/follow/`)
    .then((res) => {
      dispatch(UnfollowSuccess(res.data));
      dispatch(pageLoadedAction());
      dispatch(fetchUsersAction(user.username, listTypes.FOLLOWING));
      dispatch(fetchUsersAction(user.username, listTypes.FOLLOWERS));
    })
    .catch((errors) => {
      dispatch(UnfollowFailure(errors.response));
      dispatch(pageLoadedAction());
    });
};
