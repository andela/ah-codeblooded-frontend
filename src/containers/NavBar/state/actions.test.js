import configureStore from 'redux-mock-store';
import { pageLoadedAction, pageLoadingAction } from './actions';

const mockStore = configureStore();
const store = mockStore();

describe('page loading actions', () => {
  beforeEach(() => {
    store.clearActions();
  });

  it('dispatches page loading then page loaded', () => {
    store.dispatch(pageLoadingAction());
    store.dispatch(pageLoadedAction());
    expect(store.getActions()).toMatchSnapshot();
  });
});
