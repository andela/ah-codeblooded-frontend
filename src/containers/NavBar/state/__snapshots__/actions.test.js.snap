// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page loading actions dispatches page loading then page loaded 1`] = `
Array [
  Object {
    "type": "PAGE_LOADING",
  },
  Object {
    "type": "PAGE_LOADED",
  },
]
`;
